### PR TITLE
Add documentation for using a custom CNI

### PR DESCRIPTION
**Pull request by chrisdoherty4**
*Issue #, if available:* https://github.com/aws/eks-anywhere/issues/5217

*Description of changes:*

Add documentation for using custom CNIs.

### CODE REVIEW

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 1]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
+      cidrBlocks:
+      - 10.96.0.0/12
+    cniConfig:
+      cilium: 
+        skipUpgrade: true
+```
+
+{{% alert title="Warning" color="warning" %}}
+Clusters created using the Full Lifecycle Controller prior to v0.15 that have removed the EKS-A
+Cilium CNI must manually populate their `Cluster` object with the following annotation to ensure
```
Comment (csplinter) [inline docs/content/en/docs/reference/clusterspec/optional/cni.md:169]: How do they manually do this? via kubectl -f apply with an updated cluster spec?

Comment (chrisdoherty4) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md:169]: There's multiple ways to do it. Do we need to prescribe how? We could be clearer about the object they're editing though given there are multiple Cluster objects.

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 2]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
 2. Switching from `always` mode: When switching from `always` to `default` mode, EKS Anywhere
 will not delete any of the existing NetworkPolicy objects, including the ones required
    for EKS Anywhere components (listed above). The user must delete NetworkPolicy objects as needed.
-   
+
+### Use a custom CNI
+
+EKS-A can be configured to skip EKS-A Cilium CNI upgrades via the `skipUpgrade` field. `skipUpgrade`
```
[the remark this answers is not in the document]
Comment (csplinter) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md:140]: Just here for starters

File: docs/content/en/docs/reference/clusterspec/optional/cni.md (hunk 1 above)
Comment (csplinter) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md:169]: This sentence may not be clear to newer users. I'm assuming the annotation goes on the `Cluster` spec and I apply that via my choice of K8s API conformant tool (GitOps, Terraform, kubectl, etc.).

File: docs/content/en/docs/reference/clusterspec/optional/cni.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md:140]: I was trying to be explicit about it being EKS-A's Cilium as opposed to a default generic Cilium (users have historically struggled to appreciate that).

Does the current wording have ambiguity? Does the suggestion clear change that at all?

Either way, I think we need to optimize for clarity whatever that is. Let me know your thoughts and what you'd prefer.

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 3]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
-   
+
+### Use a custom CNI
+
+EKS-A can be configured to skip EKS-A Cilium CNI upgrades via the `skipUpgrade` field. `skipUpgrade` 
+can be `true` or `false`. When not set, it defaults to `false`.
+
+When creating a new cluster with `skipUpgrade` enabled, EKS-A Cilium will be installed. When the 
+cluster successfully provisions, EKS-A Cilium may be uninstalled and replaced with a different CNI.
+Subsequent upgrades to the cluster will not attempt to upgrade or re-install EKS-A Cilium.
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/optional/cni.md:145]:
```suggestion
Subsequent upgrades to the cluster will not attempt to upgrade or re-install EKS Anywhere Cilium.
```

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 4]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
+    services:
+      cidrBlocks:
+      - 10.96.0.0/12
+    cniConfig:
+      cilium: 
+        skipUpgrade: true
+```
+
+{{% alert title="Warning" color="warning" %}}
+Clusters created using the Full Lifecycle Controller prior to v0.15 that have removed the EKS-A
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/optional/cni.md:168]:
```suggestion
Clusters created using the Full Lifecycle Controller prior to v0.15 that have removed the EKS Anywhere
```

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 5]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
+Cilium CNI must manually populate their `Cluster` object with the following annotation to ensure
+EKS-A does not attempt to re-install EKS-A Cilium.
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/optional/cni.md:170]:
```suggestion
EKS Anywhere does not attempt to re-install EKS Anywhere Cilium.
```

**Review by chrisnegus — COMMENTED** [review]
In our docs, we try to avoid "EKS-A" and use the formal name "EKS Anywhere". Otherwise, the content looks good to me.

File: docs/content/en/docs/reference/clusterspec/optional/cni.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (csplinter) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md:140]: I think we need a page in our docs that spells out exactly what's in EKS-A's Cilium vs OSS Cilium vs Enterprise Cilium. but for now let's just leave it at what you have, I think it's fine until we have a docs page with more clarity to point to.

[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md]: Updated as suggested.

File: docs/content/en/docs/reference/clusterspec/optional/cni.md [hunk 6]
```diff
@@ -134,7 +134,46 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
 will not delete any of the existing NetworkPolicy objects, including the ones required
    for EKS Anywhere components (listed above). The user must delete NetworkPolicy objects as needed.
-   
+
+### Use a custom CNI
+
+EKS-A can be configured to skip EKS-A Cilium CNI upgrades via the `skipUpgrade` field. `skipUpgrade` 
+can be `true` or `false`. When not set, it defaults to `false`.
+
+When creating a new cluster with `skipUpgrade` enabled, EKS-A Cilium will be installed. When the
```
[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md]: Added the elaboration.

File: docs/content/en/docs/reference/clusterspec/optional/cni.md (hunk 1 above)
Comment (chrisdoherty4) [reply docs/content/en/docs/reference/clusterspec/optional/cni.md]: Made clear its a cluster spec object and the exact object that needs annotating.

**Pull request merged**